src/x11_utils.rs

```diff
@@ -525,3 +526,7 @@ macro_rules! __atom_manager_atom_value {
         $atom_value
     };
 }
+
+pub(crate) fn cow_strip_length(cow: Cow<'_, [u8; 32]>) -> Cow<'static, [u8]> {
+    Cow::Owned(cow.into_owned().to_vec())
```

Comment:
If I'm not wrong, `cow.into_owned()` is returning a `Box<[u8; 32]>`, and `.to_vec` is actually allocating memory and copying the bytes to create the `Vec`. I think it would be better to do:
```rust
let boxed_slice: Box<[u8]> = cow.into_owned();
Cow::Owned(boxed_slice.into())
```
The `let` will coerce the `Box<[u8; 32]>` into a `Box<[u8]>` and `.into()` will convert the `Box<[u8]>` into a `Vec<u8>` reusing the allocation.

Reply:
`Cow.into_owned` returns `[u8; 32]` here not `Box<[u8; 32]>`